torch/distributed/_spmd/data_parallel.py

```diff
@@ -367,7 +368,10 @@ def build_data_parallel_strategies(
                     input_nodes[0]
                 ].reduction_over_batch
 
-                if arg_node_type == NodeType.PARAM:
+                if node.target == operator.getitem:
+                    # for getitem call, just forward the strategy from the input
+                    dp_strategy_map[node] = dp_strategy_map[node.args[0]]
```

Comment:
hmm, input here is a list of nodes and output might be a list or tensor. So do we mark the foreach output as a PARAM or NON_TENSOR?

Reply:
Also, does this mean individual tensors from foreach output does not have a dp strategy associated with it?

Reply:
nvm, I saw the PRs on top is trying to address this problem